### PR TITLE
Added documentation for key binding, contributions and registry

### PR DESCRIPTION
**Pull request by JonasHelming**
Contributed on behalf of STMicroelectronics

Signed-off-by: Jonas Helming <jhelming@eclipsesource.com>

#### Review checklist

- [ ] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/common/keybinding.ts [hunk 1]
```diff
@@ -13,11 +13,19 @@
  *
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
-
+/**
+ * A Keybinding binds a specific key sequence ({@link keybinding}) to trigger a command ({@link command}). A Keybinding optionally may
```
Comment (tortmayr) [inline packages/core/src/common/keybinding.ts:17]:
```suggestion
 * A Keybinding binds a specific key sequence ({@link Keybinding#keybinding}) to trigger a command ({@link Keybinding#command}). A Keybinding optionally may
```

File: packages/core/src/common/keybinding.ts [hunk 2]
```diff
@@ -13,11 +13,19 @@
...
+ * define a "when clause" ({@link when}) to specific in which context it becomes active.
```
Comment (tortmayr) [inline packages/core/src/common/keybinding.ts:18]:
```suggestion
 * define a "when clause" ({@link Keybinding#when}) to specify in which context it becomes active.
```

File: packages/core/src/common/keybinding.ts [hunk 3]
```diff
@@ -28,12 +36,14 @@ export interface Keybinding {
      */
     context?: string;
     /**
-     * https://code.visualstudio.com/docs/getstarted/keybindings#_when-clause-contexts
+     * A clause defining the condition when the Keybinding is active, e.g. based on the current focus.
```
Comment (koegel) [inline packages/core/src/common/keybinding.ts:39]:
```suggestion
     * A optional clause defining the condition when the Keybinding is active, e.g. based on the current focus.
```

File: packages/core/src/common/keybinding.ts [hunk 4]
```diff
@@ -28,12 +36,14 @@ export interface Keybinding {
...
+     * See {@link https://code.visualstudio.com/docs/getstarted/keybindings#_when-clause-contexts} for more details.
      */
     when?: string;
 
     /**
-     * Specified when the command has arguments that are passed to the command handler.
+     * Arguments that will be passed to the command when it gets triggered via this KeyBinding.
```
Comment (koegel) [inline packages/core/src/common/keybinding.ts:45]:
```suggestion
     * Optional arguments that will be passed to the command when it gets triggered via this KeyBinding.
```

**Review by koegel — CHANGES_REQUESTED** [review]
Minor changes

**Review by tortmayr — APPROVED** [review]

**Comment by sdirix** [comment]
Merged upstream with https://github.com/eclipse-theia/theia/pull/8637

**Pull request closed without merging**